core-common/src/main/java/org/mapstruct/AfterMapping.java

```diff
@@ -0,0 +1,103 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a method to be invoked at the end of a generated mapping method, right before the last {@code return} statement
+ * of the mapping method. The method can be implemented in an abstract mapper class or be declared in a type (class or
+ * interface) referenced in {@link Mapper#uses()} in order to be used in a mapping method.
+ * <p>
+ * Only methods with return type {@code void} may be annotated with this annotation.
```

Comment:
`@BeforeMapping` and `@AfterMapping` are both treated the same when it comes to parameter matching. On the one hand, the parameters of the lifecycle methods (in combination with `@MappingTarget` and `@TargetType`) are used to have control over the situations in which these methods are pulled in and applied. On the other hand, you are free to use these values to perform what ever you intended.
For example, an `@AfterMapping` method could be used to add some manual post-processing to the mapping target, using values form the mapping source (we have such a case at work, with abstract source and target types).